hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFilePreadReader.java

```diff
@@ -37,6 +38,17 @@ public class HFilePreadReader extends HFileReaderImpl {
   public HFilePreadReader(ReaderContext context, HFileInfo fileInfo, CacheConfig cacheConf,
     Configuration conf) throws IOException {
     super(context, fileInfo, cacheConf, conf);
+
+    try {
+      DataTieringManager dataTieringManager = DataTieringManager.getInstance();
+      if (!dataTieringManager.isHotData(this)) {
+        LOG.debug("Data tiering is enabled for path '{}' and it is not hot data", path);
+        return;
+      }
+    } catch (IllegalStateException e) {
+      LOG.error("Error while getting DataTieringManager instance: {}", e.getMessage());
```

Comment:
Please move this to BucketCache.shouldCacheFile implementation. Any caching related logic should be in the Cache related classes. Reader should only defer the decisions about caching to the Cache implementations.

Reply:
Another option is something like below.
```
public HFilePreadReader(ReaderContext context, HFileInfo fileInfo, CacheConfig cacheConf,
    Configuration conf) throws IOException {
    super(context, fileInfo, cacheConf, conf);

    boolean isHotData = true;
    try {
      DataTieringManager dataTieringManager = DataTieringManager.getInstance();
      isHotData = dataTieringManager.isHotData(this);
    } catch (IllegalStateException e) {
      LOG.error("Error while getting DataTieringManager instance: {}", e.getMessage());
    }

    final MutableBoolean shouldCache = new MutableBoolean(true);

    boolean finalIsHotData = isHotData;
    cacheConf.getBlockCache().ifPresent(cache -> {
      Optional<Boolean> result = cache.shouldCacheFile(path.getName(), finalIsHotData);
      shouldCache.setValue(result.isPresent() ? result.get().booleanValue() : true);
    });
```

Reply:
Yeah, this looks correct. but we also need Configuration which is not present in HFileInfo.